stage0/run.go

```diff
@@ -452,7 +452,7 @@ func prepareStage1Image(cfg PrepareConfig, img types.Hash, cdir string, useOverl
 func setupStage1Image(cfg RunConfig, img types.Hash, cdir string, useOverlay bool) error {
 	if useOverlay {
 		s1 := common.Stage1ImagePath(cdir)
-		if err := overlayRender(cfg, img, cdir, s1); err != nil {
+		if err := overlayRender(cfg, img, cdir, s1, "stage1"); err != nil {
```

Comment:
The upper directory still contains the image hash. The `appName` is just an additional suffix. In your example, `example.com/totally/not/rkt/stage1` and the real stage1 will have different image hash, so the upper directories will be different, even if they share the same suffix `/stage1`.